pkg/installer/uninstall.go

```diff
@@ -231,6 +248,32 @@ func (in *Installer) uninstallEtcdOperator() error {
 	return err
 }
 
+// checkForExistingWorkloads ensures no bound PVCs are using a storageos storageclass.
+func (in *Installer) checkForExistingWorkloads() error {
+	pvcList, err := pluginutils.ListPersistentVolumeClaims(in.clientConfig, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	scList, err := pluginutils.ListStorageClasses(in.clientConfig, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, pvc := range pvcList.Items {
+		if pvc.Status.Phase != corev1.ClaimBound {
+			continue
+		}
+		if provisioner, ok := pvc.GetAnnotations()["volume.beta.kubernetes.io/storage-provisioner"]; ok && provisioner == stosSCProvisioner {
+			return fmt.Errorf(errWorkloadsExist, pvc.Name, *pvc.Spec.StorageClassName)
+		}
+		for _, sc := range scList.Items {
+			if sc.Name == *pvc.Spec.StorageClassName && sc.Provisioner == stosSCProvisioner {
+				return fmt.Errorf(errWorkloadsExist, pvc.Name, *pvc.Spec.StorageClassName)
+			}
+		}
```

Comment:
Personally, I'd create separate functions for the annotation check and the SC check, and then have an `IsStorageOSPVC(PVC *corev1.PersistentVolumeClaim) (bool, error)`  that can check each bound PVC.